aya/src/programs/links.rs

```diff
@@ -369,6 +369,180 @@ pub enum LinkError {
     SyscallError(#[from] SyscallError),
 }
 
+/// A [`Link`] identifier, which may or may not be owned by aya.
+pub struct LinkId(u32);
+
+impl LinkId {
+    /// A wrapper for an arbitrary loaded link specified by its kernel id,
+    /// unsafe because there is no guarantee provided by aya that the link is
+    /// still loaded or even exists.
+    pub unsafe fn new(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+bitflags::bitflags! {
+    /// Flags which are use to build a set of MprogOptions.
```

Comment:
What are MprogOptions?

Reply:
https://github.com/torvalds/linux/blob/master/include/uapi/linux/bpf.h#L1201 Thee new genric multi-program api flags, we follow the same pattern here -> https://github.com/aya-rs/aya/blob/main/aya/src/programs/xdp.rs#L43